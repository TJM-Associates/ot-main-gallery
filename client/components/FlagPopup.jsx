import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Container = styled.div`
  z-index: 3;
  position: fixed;
  background-color: white;
  display: flex;
  flex-direction: column;
  width: 400px;
  height: 245.406px;
  left: 50%;
  top: 200px;
  margin-left: -200px;
`;

Container.displayName = 'Container';

const Button = styled.div`
  flex: 1;
  text-align: center;
  vertical-align: middle;
  background-color: #61bddb;
  color: white;
  margin: 2.5px 17px;
  border-radius: 3px;
  padding-top: 12px;
  cursor: pointer;
`;

Button.displayName = 'Button';

const CancelButton = styled(Button)`
  background-color: white;
  color: #61bddb;
`;

CancelButton.displayName = 'CancelButton';

const Header = styled.div`
  flex: 1;
  font-size: 20px;
  font-weight: strong;
  padding: 10px;
  border-bottom: solid 0.5px #DCDCDC;
  margin-bottom: 2px;
`;

Header.displayName = 'Header';

const FlagPopup = ({ closeFlagPopup }) => {
  const handleClick = () => {
    console.log('clicked');
    closeFlagPopup();
  };

  return (
    <Container>
      <Header>Report a photo problem</Header>
      <Button onClick={handleClick}>Unrelated to restaurant</Button>
      <Button onClick={handleClick}>Inappropriate content</Button>
      <Button onClick={handleClick}>I don't like this photo</Button>
      <CancelButton onClick={handleClick}>Cancel</CancelButton>
    </Container>
  );
};

FlagPopup.propTypes = {
  closeFlagPopup: PropTypes.func,
};

FlagPopup.defaultProps = {
  closeFlagPopup: () => {},
};

export default FlagPopup;
