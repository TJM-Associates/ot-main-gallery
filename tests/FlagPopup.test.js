import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import FlagPopup from '../client/components/FlagPopup.jsx';

describe('FlagPopup', () => {
  it('renders successfully', () => {
    const wrapper = shallow(<FlagPopup />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('Container').length).toEqual(1);
    expect(wrapper.find('Button').length).toEqual(3);
    expect(wrapper.find('Header').length).toEqual(1);
    expect(wrapper.find('CancelButton').length).toEqual(1);
  });
  it('stops rendering when any of the buttons are clicked', () => {
    const wrapper = shallow(<FlagPopup />);
    wrapper.find('Button').first().simulate('click');
    console.log(wrapper.find('Button').length);
  });
});
