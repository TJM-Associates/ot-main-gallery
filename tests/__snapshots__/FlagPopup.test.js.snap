// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlagPopup renders successfully 1`] = `
<Container>
  <Header>
    Report a photo problem
  </Header>
  <Button
    onClick={[Function]}
  >
    Unrelated to restaurant
  </Button>
  <Button
    onClick={[Function]}
  >
    Inappropriate content
  </Button>
  <Button
    onClick={[Function]}
  >
    I don't like this photo
  </Button>
  <CancelButton
    onClick={[Function]}
  >
    Cancel
  </CancelButton>
</Container>
`;
